diesel/src/expression/mod.rs

```diff
@@ -544,12 +544,18 @@ where
 /// # }
 /// ```
 ///
-/// If you want to avoid nesting types, you can use the
-/// [`Selectable`](derive@Selectable) derive macro's
-/// `select_expression` and `select_expression_type` attributes to
-/// flatten the fields. If the `select_expression` is simple enough,
-/// it is not necessary to specify `select_expression_type`
-/// (most query fragments are supported for this).
+/// It is also possible to specify an entirely custom select expression
+/// for fields when deriving [`Selectable`](derive@Selectable).
+/// This is useful for example to
+///
+///  * avoid nesting types, or to
+///  * populate fields with values other than table columns, such as
+///    the result of an SQL function like `CURRENT_TIMESTAMP()`
+///    or a custom SQL function.
+///
+/// The select expression is specified via the `select_expression` parameter.
+/// For more complex expressions, it may be required to also specify the result type
+/// of the expression with `select_expression_type` (most query fragments are supported for this).
```

Comment:
I feel the last part in the parenthesis needs to be updated now. Something like the following should work.

```suggestion
/// of the expression with `select_expression_type`. For most query fragments this can be inferred and is not needed to be specified.
```

Reply:
Done.